File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        // int j=1;
        string res="";
        for(int i=0;i<strs[0].size();i++){
            int ch=strs[0][i];
            for(int j=1;j<strs.size();j++){
                if(i>=strs[j].size() || strs[j][i]!=ch){
                    return res;
                }
                
            }
            res+=ch;
        }
        return res;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String st="";
        for(int i=0;i<strs[0].length();i++){
            char c=strs[0].charAt(i);
            int j=1;
            while(j<strs.length){
                if(i>=strs[j].length() || strs[j].charAt(i)!=c){
                    return st;
                
                }
                j++;
            }
            st+=c;
        }
        return st;
    }
}